lib/trie.js

```diff
@@ -0,0 +1,66 @@
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {};
+}
+
+//insert new nodes to Trie
+Trie.prototype.add = function(path, verb, handler) {
+  var node = this;
+  var parts = getPathParts(path);
+  if (parts.length) {
+    for (var i = 0; i < parts.length; i++) {
+      if (!(parts[i] in node)) {
+        node[parts[i]] = new Trie();
+      }
+      if (i + 1 === parts.length) {
+        addVerbHandler(node[parts[i]], verb, handler);
+      }
+      node = node[parts[i]];
+    }
+  } else { // if we would ever encounter this situation?
+    addVerbHandler(node, verb, handler);
+  }
+  return this;
+};
+
+//Look for matching path and return handler method for a match
+Trie.prototype.find = function(path) {
+  var node = this;
+  var parts = getPathParts(path);
+  for (var i = 0; i < parts.length; i++) {
+    if (!(parts[i] in node)) {
+      return false;
+    } else {
+      if (i + 1 === parts.length)
+        return node[parts[i]].methods;
+    }
+    node = node[parts[i]];
+  }
+};
+
+function getPathParts(path) {
+  path = path.trim();
+  var chunks = path.split('/');
+  var parts = [];
+  chunks.forEach(function(chunk) {
+    if (chunk) {
+      parts.push(chunk);
+    }
+  });
```

Comment:
This can be simplified to `var parts = chunks.filter(function(c) { return !!c; })`.

However, we have learned that `forEach` (or `filter`) are better avoided in code paths frequently invoked, as they incur performance overhead due to the need to allocate a closure for the callback function.

Please rewrite this method to use `for` instead.